### PR TITLE
narzedzia#2

### PR DESCRIPTION
**Pull request by marcindus**

### CODE REVIEW

File: functions.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+void sort(std::vector<int> & unsorted) {
+    std::sort(begin(unsorted), end(unsorted));
+}
+
+void reverse(std::vector<int> & data) {
+    std::reverse(begin(data), end(data));
+}
+
```
Comment (Marcinati) [inline functions.cpp:19]: We should merge this file with src/functions.hpp and remove first one.

**Review by Marcinati — COMMENTED** [review]
Files structure from master branch should be preserved.

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 src/*.cpp -o main -Wall -Werror -Iinc
```
Comment (DamiXioR) [inline Makefile:2]: it would look better in this way: -I./inc

**Review by DamiXioR — COMMENTED** [review]
you forgot remove functions.cpp file from reverse branch

File: src/functions.cpp [hunk 3]
```diff
@@ -1,10 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
 #include <iostream>
 #include <limits>
+#include <algorithm>
+#include "functions.hpp"
```
Comment (Marcinati) [inline src/functions.cpp:4]: Sort headers.

File: tutor [hunk 4]
```diff
@@ -0,0 +1,970 @@
...
+
+  3. Move the cursor to the beginning of a word that needs to be deleted.
+
+  4. Type   dw	 to make the word disappear.
+
+  NOTE: The letter  d  will appear on the last line of the screen as you type
+	it.  Vim is waiting for you to type  w .  If you see another character
+	than  d  you typed something wrong; press  <ESC>  and start over.
+
+---> There are a some words that don't belong paper in this sentence.
```
Comment (ziobron) [inline tutor:203]: not completed

File: tutor [hunk 5]
```diff
@@ -0,0 +1,970 @@
...
+
+  1. Press  <ESC>  to make sure you are in Normal mode.
+
+  2. Move the cursor to the line below marked --->.
+
+  3. Move the cursor to the end of the correct line (AFTER the first . ).
+
+  4. Type    d$    to delete to the end of the line.
+
+---> Somebody typed the end of this line twice. end of this line twice.
```
Comment (ziobron) [inline tutor:222]: not completed

File: tutor [hunk 6]
```diff
@@ -0,0 +1,970 @@
...
+     first error.
+  2. Type  x  to delete the first unwanted character.
+  3. Now type  u  to undo the last command executed.
+  4. This time fix all the errors on the line using the  x  command.
+  5. Now type a capital  U  to return the line to its original state.
+  6. Now type  u  a few times to undo the  U  and preceding commands.
+  7. Now type CTRL-R (keeping CTRL key pressed while hitting R) a few times
+     to redo the commands (undo the undo's).
+
+---> Fiix the errors oon and reeplace them witth undo.
```
Comment (ziobron) [inline tutor:336]: not completed

**Comment by ziobron** [comment]
Anyway max points ;)

+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**